### DISCUSSION TITLE
linkToFileDownload - Скачать файлы

### DISCUSSION THREAD

**Discussion by Makhmut2445** [totumonline/totum-ru-issues-and-discussions#217] [Как это сделать?] [answered]
Как заполнить правильно, чтоб можно было скачать файлы во всех строках по полю типа файл? 
Тип файла .pdf. 
Поле не множественное. 
Таблица Цикловая.

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Не понял задачу. Есть несколько строк, в них файлы и надо скачать все файлы например по колонке?

**Reply by Makhmut2445** [reply]
> Не понял задачу. Есть несколько строк, в них файлы и надо скачать все файлы например по колонке?

Да

**Reply by Makhmut2445** [reply]
Можете помочь?

**Reply by ttm-alexeypolunin** [reply]
```
= : linkToFileDownload(files: $get_content)

get_content: listReplace(list: $remove; action: "filestring" = $val; value: "value")
	
		val: fileGetContent(file: $#value[filestring])
	
	remove: rowKeysRemove(row: $replace; key: "size"; recursive: true)
		replace: rowKeysReplace(row: $train; from: json`["ext","file"]`; to: json`["type","filestring"]`; recursive: true)
			train: listTrain(list: $sel)
				sel: selectList(table: $#ntn; field: 'file_field')
```

**Reply by Makhmut2445** [reply]
Как снять ограничение?
Поставил условие: where: 'id' = $#ids
Выбираю 50 файлов ,скачивает только 14.
Выбираю 20 файлов ,скачивает только 17.

**Reply by ttm-alexeypolunin** [reply]
Есть ограничение у PHP на количество одновременно загружаемых — https://www.php.net/manual/ru/ini.core.php#ini.max-file-uploads, возможно оно влияет и на скачку.

**Reply by ttm-alexeypolunin** [reply]
А вообще бы делать через execSSH упаковку в .zip и скачку одного файла.

**Reply by Makhmut2445** [reply]
> А вообще бы делать через execSSH упаковку в .zip и скачку одного файла.

Так было бы идеально, вот только знать бы как это делается..

**Reply by ttm-alexeypolunin** [reply]
Будет добавлен параметр в `linkToFileDownload` https://github.com/totumonline/totum-ru-issues-and-discussions/discussions/220
